src/Backends/SystemUpdate.vala

```diff
@@ -0,0 +1,255 @@
+    public enum State {
+        UP_TO_DATE,
+        CHECKING,
+        AVAILABLE,
+        DOWNLOADING,
+        RESTART_REQUIRED,
+        ERROR
+    }
+
+    public struct CurrentState {
+        State state;
+        string status;
+    }
+
+    public struct UpdateDetails {
+        string[] packages;
+        int size;
+    }
+
+    public signal void state_changed ();
+
+    private static Settings settings = new GLib.Settings ("io.elementary.settings-daemon.system-updates");
+
+    private CurrentState current_state;
+    private UpdateDetails update_details;
+
+    private Pk.Task task;
+    private Pk.PackageSack? available_updates = null;
+    private GLib.Cancellable cancellable;
+
+    construct {
+        current_state = {
+            UP_TO_DATE,
+            ""
+        };
+
+        update_details = {
+            {},
+            0
+        };
+
+        task = new Pk.Task () {
+            only_download = true
+        };
+
+        cancellable = new GLib.Cancellable ();
+
+        check_for_updates.begin (false, true);
+
+        Timeout.add_seconds ((uint) settings.get_int64 ("refresh-interval"), () => {
+            check_for_updates.begin (false, true);
+            return Source.CONTINUE;
+        });
+    }
+
+    public async void check_for_updates (bool force, bool notify) throws DBusError, IOError {
+        if (current_state.state != UP_TO_DATE && current_state.state != AVAILABLE && !force) {
+            return;
+        }
+
+        update_state (CHECKING);
+
+        try {
+            yield task.refresh_cache_async (force, null, progress_callback);
+        } catch (Error e) {
+            warning ("Failed to refresh cache: %s", e.message);
+        }
+
+        try {
+            available_updates = (yield task.get_updates_async (Pk.Filter.NONE, null, progress_callback)).get_package_sack ();
+
+            settings.set_int64 ("last-refresh-time", new DateTime.now_utc ().to_unix ());
+
+            if (available_updates == null || available_updates.get_size () == 0) {
+                update_state (UP_TO_DATE);
+                return;
+            }
+
+            string[] package_names = {};
+
+            foreach (var package in available_updates.get_array ()) {
+                package_names += package.get_name ();
+            }
+
+            update_details = {
+                package_names,
+                0 //FIXME: Is there a way to get update size from PackageKit
+            };
+
+            if (notify) {
+                var notification = new Notification (_("Update Available"));
+                notification.set_body (_("A system update is available"));
+                notification.set_icon (new ThemedIcon ("software-update-available"));
+                notification.set_default_action (Application.ACTION_PREFIX + Application.SHOW_UPDATES_ACTION);
+
+                GLib.Application.get_default ().send_notification (null, notification);
```

Comment:
We should set an ID here so that these notifications replace each other and don't stack up duplicates in the notification center
```suggestion
                GLib.Application.get_default ().send_notification ("system-update", notification);
```